### PR TITLE
Fix S2589  FP: FlowCaptureReference in throw statement

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**

### CODE REVIEW

**Review by mary-georgiou-sonarsource — COMMENTED** [review]
Looks ok - waiting for pipeline + a second opinion if possible too.

EDIT: ITs are failing.

File: analyzers/its/expected/Automapper/AutoMapper--net461-S2589.json [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-"location":  {
-"uri":  "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/sources/AutoMapper/src/AutoMapper/Configuration/ConfigurationValidator.cs#L127",
-"region":  {
-"startLine":  127,
-"startColumn":  27,
-"endLine":  127,
-"endColumn":  114
-}
-}
-}
```
Comment (Tim-Pohlmann) [inline analyzers/its/expected/Automapper/AutoMapper--net461-S2589.json:15]: FP

File: analyzers/its/expected/Automapper/AutoMapper--netstandard2.0-S2589.json [hunk 2]
```diff
@@ -1,17 +0,0 @@
...
-"location":  {
-"uri":  "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/sources/AutoMapper/src/AutoMapper/Configuration/ConfigurationValidator.cs#L127",
-"region":  {
-"startLine":  127,
-"startColumn":  27,
-"endLine":  127,
-"endColumn":  114
-}
-}
-}
```
Comment (Tim-Pohlmann) [inline analyzers/its/expected/Automapper/AutoMapper--netstandard2.0-S2589.json:15]: FP

File: analyzers/its/expected/Nancy/Nancy--net452-S2589.json [hunk 3]
```diff
@@ -25,19 +25,6 @@
...
-"message":  "Change this condition so that it does not always evaluate to 'True'.",
-"location":  {
-"uri":  "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/sources/Nancy/src/Nancy/ViewEngines/DefaultViewFactory.cs#L87",
-"region":  {
-"startLine":  87,
-"startColumn":  23,
-"endLine":  87,
-"endColumn":  174
-}
-}
```
Comment (Tim-Pohlmann) [inline analyzers/its/expected/Nancy/Nancy--net452-S2589.json:40]: FP

File: analyzers/its/expected/Nancy/Nancy--netstandard2.0-S2589.json [hunk 4]
```diff
@@ -25,19 +25,6 @@
...
-"message":  "Change this condition so that it does not always evaluate to 'True'.",
-"location":  {
-"uri":  "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/sources/Nancy/src/Nancy/ViewEngines/DefaultViewFactory.cs#L87",
-"region":  {
-"startLine":  87,
-"startColumn":  23,
-"endLine":  87,
-"endColumn":  174
-}
-}
```
Comment (Tim-Pohlmann) [inline analyzers/its/expected/Nancy/Nancy--netstandard2.0-S2589.json:40]: FP

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**